[functorch] delete functorch/csrc/Constants.h

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #84648
* #84642
* __->__ #84639

This file aliased dispatch keys. The original purpose was so that we
could change the dispatch keys in pytorch core without changing
functorch too much, but there's no need for the layer of indirection
anymore.

Also moved SINGLE_ARG to functorch/csrc/Macros.h, but that might need a
new home later.

Differential Revision: [D39315052](https://our.internmc.facebook.com/intern/diff/D39315052)

zou3519 committed:
[functorch] delete some macros

These macros aliased dispatch keys but used the same or more characters
so they served no real purpose.

[ghstack-poisoned]

zou3519 committed:
Update on "[functorch] delete some macros"


This PR deletes the following macros (that expand to dispatch keys):
- FT_BATCHED
- FT_VMAP_MODE
- FT_GRAD_WRAPPER
- FT_DYNAMIC_LAYER_FRONT_MODE
- FT_DYNAMIC_LAYER_BACK_MODE
These macros aliased dispatch keys but used the same or more characters
so they served no real purpose.

[ghstack-poisoned]

zou3519 committed:
Update on "[functorch] delete some macros"


This PR deletes the following macros (that expand to dispatch keys):
- FT_BATCHED
- FT_VMAP_MODE
- FT_GRAD_WRAPPER
- FT_DYNAMIC_LAYER_FRONT_MODE
- FT_DYNAMIC_LAYER_BACK_MODE
These macros aliased dispatch keys but used the same or more characters
so they served no real purpose.

[ghstack-poisoned]

zou3519 committed:
Update on "[functorch] delete some macros"


This PR deletes the following macros (that expand to dispatch keys):
- FT_BATCHED
- FT_VMAP_MODE
- FT_GRAD_WRAPPER
- FT_DYNAMIC_LAYER_FRONT_MODE
- FT_DYNAMIC_LAYER_BACK_MODE
These macros aliased dispatch keys but used the same or more characters
so they served no real purpose.

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39315052).

samdow commented on functorch/functorch/csrc/Constants.h:
```diff
@@ -8,26 +8,16 @@
-constexpr auto kDynamicLayerFrontModeKey = c10::DispatchKey::FT_DYNAMIC_LAYER_FRONT_MODE_KEY;
-constexpr auto kDynamicLayerBackModeKey = c10::DispatchKey::FT_DYNAMIC_LAYER_BACK_MODE_KEY;
-//# constexpr auto kPythonKey = c10::DispatchKey::FT_PYTHON_KEY;
+constexpr auto kBatchedKey = c10::DispatchKey::FuncTorchBatched;
```
Why do we have these since it looks like now we're using the actual dispatch name (i.e. "FuncTorchBatched" instead of "kBatchedKey")?

zou3519 replied:
`kBatchedKey` is shorter than typing out `DispatchKey::FuncTorchBatched`, though I guess not by far. Do you think we should just go with `DispatchKey::FuncTorchBatched` ?

zou3519 committed:
Update on "[functorch] delete functorch/csrc/Constants.h"


This file aliased dispatch keys. The original purpose was so that we
could change the dispatch keys in pytorch core without changing
functorch too much, but there's no need for the layer of indirection
anymore.

Also moved SINGLE_ARG to functorch/csrc/Macros.h, but that might need a
new home later.

Differential Revision: [D39315052](https://our.internmc.facebook.com/intern/diff/D39315052)

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39315052).

zou3519 committed:
Update on "[functorch] delete functorch/csrc/Constants.h"


This file aliased dispatch keys. The original purpose was so that we
could change the dispatch keys in pytorch core without changing
functorch too much, but there's no need for the layer of indirection
anymore.

Also moved SINGLE_ARG to functorch/csrc/Macros.h, but that might need a
new home later.

Differential Revision: [D39315052](https://our.internmc.facebook.com/intern/diff/D39315052)

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39315052).

zou3519 committed:
Update on "[functorch] delete functorch/csrc/Constants.h"


This file aliased dispatch keys. The original purpose was so that we
could change the dispatch keys in pytorch core without changing
functorch too much, but there's no need for the layer of indirection
anymore.

Also moved SINGLE_ARG to functorch/csrc/Macros.h, but that might need a
new home later.

Differential Revision: [D39315052](https://our.internmc.facebook.com/intern/diff/D39315052)

[ghstack-poisoned]

zou3519 committed:
Update on "[functorch] delete functorch/csrc/Constants.h"


This file aliased dispatch keys. The original purpose was so that we
could change the dispatch keys in pytorch core without changing
functorch too much, but there's no need for the layer of indirection
anymore.

Also moved SINGLE_ARG to functorch/csrc/Macros.h, but that might need a
new home later.

Differential Revision: [D39315052](https://our.internmc.facebook.com/intern/diff/D39315052)

[ghstack-poisoned]

zou3519 committed:
Update on "[functorch] delete functorch/csrc/Constants.h"


This file aliased dispatch keys. The original purpose was so that we
could change the dispatch keys in pytorch core without changing
functorch too much, but there's no need for the layer of indirection
anymore.

Also moved SINGLE_ARG to functorch/csrc/Macros.h, but that might need a
new home later.

Differential Revision: [D39315052](https://our.internmc.facebook.com/intern/diff/D39315052)

[ghstack-poisoned]
